openpathsampling/treelogic.py

```diff
@@ -1,169 +1,247 @@
+        """
+        return hex(id(self))
 
+    @property
+    def _choices(self):
+        if self._node_type == self.NODE_TYPE_ALL:
+            return [[sub for sub in self._subnodes]]
+        elif self._node_type == self.NODE_TYPE_ACCUMULATE:
+            return [self._subnodes[:n+1] for n in range(0, len(self._subnodes))]
+        elif self._node_type == self.NODE_TYPE_ONE:
+            leaves = [[sub] for sub in self._subnodes]
+
+            unique_leaves = list(set([ tuple(l) for l in leaves ]))
+
+            return unique_leaves
+        elif self._node_type == self.NODE_TYPE_POWER:
+            s = list(self._subnodes)
+            return itertools.chain.from_iterable(
+                itertools.combinations(s, r) for r in range(len(s)+1)
+            )
+        elif self._node_type == self.NODE_TYPE_NONE:
+            return []
+        elif hasattr(self._node_type, '__call__'):
+            return self._node_type()
+        else:
+            raise RuntimeError('Node has no type !')
 
-    def __reversed__(self):
-        """
-        Traverse the whole tree in post-order
+    def treeprint(self):
+        return str(self.head) + "\n" + TreeSetMixin._indent("\n".join(map(lambda x : x.treeprint(), self.branches)))
 
-        Returns
-        -------
-        Iterator
-            an iterator that traverses the tree in post-order
-        """
-        for subchange in self._subnodes:
-            for change in reversed(subchange):
-                yield change
+    def locate(self, item):
+        l = [key for key, value in self.locators().iteritems() if self._default_match(value, item)]
+        if len(l) == 0:
+            return None
+        elif len(l) == 1:
+            return l[0]
+        else:
+            return l
+
+    def pick(self, item):
+        loc = self.locate(item)
+        if loc is None:
+            return loc
+        elif type(loc) is list:
+            return [self[l] for l in loc]
+        else:
+            return self[loc]
 
-        yield self
+    @property
+    def is_sequential(self):
+        return len(self._choices) < 2
 
-    def __len__(self):
-        """
-        Returns the total number of Changes mad in a move.
+    @property
+    def deterministic(self):
+        if not hasattr(self, '_deterministic'):
+            if not self.is_sequential:
+                self._deterministic = False
+            else:
+                self._deterministic = True
+                for node in self._subnodes:
+                    if not node.deterministic:
+                        self._deterministic = False
 
-        Returns
-        -------
-        int
-            the number of (Sub)PathMoveChanges in this PathMoveChange
+        return self._deterministic
 
+    @property
+    def unique(self):
         """
-        if self._len is None:
-            self._len = len(list(iter(self)))
+        Return the smallest tree of tuples that uniquely represents this tree
+        """
+        ret = []
```

Comment:
Good point. I was looking for a short name since it could be that this property is used more frequently than others. I like to use `minimal` somehow. It is the smallest possible subtree that uniquely represents a tree. I need to think about that. 
The word unique is in relation to the generating (2D) tree. So with respect to all possible outcomes of moves (trees) what is the smallest possible subtree that can only be present in this tree and in no other one. Hence it uniquely identifies this particular PMC.